### PR TITLE
Last step for allowing custom Styles

### PR DESCRIPTION
**Pull request by ravolt**
This allows engines to override the SASS directive.  NOTE: only 1 can win, so you can't do it in multiple engines.

Also, updated the docs and capture the git commit in the config.  Currently that is not exposed.

### CODE REVIEW

File: rails/config/application.rb [hunk 1]
```diff
@@ -31,6 +31,7 @@ class Application < Rails::Application
     # -- all .rb files in that directory are automatically loaded.
     config.crowbar = ActiveSupport::OrderedOptions.new 
     config.crowbar.version = '2.x'
+    config.crowbar.commit = %x[git log -n 1].split("\n")[0] rescue 'n/a'
```
Comment (galthaus) [inline rails/config/application.rb]: I'm worried that this might not fail cleanly.

**Comment by galthaus** [comment]
-1 on git log call that way.

File: rails/app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -3,7 +3,7 @@
   %head
     %title= t 'title', default: "OpenCrowbar"
     = csrf_meta_tags
-    = stylesheet_link_tag 'application'
+    = stylesheet_link_tag Rails.configuration.crowbar.sass_base || 'application'
```
Comment (ravolt) [inline rails/app/views/layouts/application.html.haml]: This is the primary target of the pull request!

**Comment by galthaus** [comment]
Retrying the test, but I think you broke the barclamp version rspec test.

File: rails/app/models/barclamp.rb [hunk 3]
```diff
@@ -55,15 +55,16 @@ def self.import(bc_name="core", bc=nil, source_path=nil, parent_id=nil)
...
       if bc['git']
-        gitcommit = bc['git']['commit'] || 'unknown'
-        gitdate = bc['git']['date'] || 'unknown'
+        gitcommit = bc['git']['commit'] || gitcommit
+        gitdate = bc['git']['date'] || gitdate
       end
-      version = bc["version"] || '2.0'
-      build_version = bc["build_version"] || '2.0'
+      version = bc["version"] || '0.0'
+      build_version = bc["build_version"] || '0.0'
```
Comment (galthaus) [inline rails/app/models/barclamp.rb]: You changed the default from 2.0 to 0.0 in a barclamp.  This broke the test.  I don't care which you change, but one of them needs to.

**Pull request merged**